contrib/database/sql/sql.go

```diff
@@ -236,3 +232,29 @@ func Open(driverName, dataSourceName string, opts ...Option) (*sql.DB, error) {
 	}
 	return OpenDB(&dsnConnector{dsn: dataSourceName, driver: d}, opts...), nil
 }
+
+func processOptions(cfg *config, driverName string, opts ...Option) {
+	for _, fn := range opts {
+		fn(cfg)
+	}
+	checkDBMPropagation(cfg, driverName)
+}
+
+func checkDBMPropagation(cfg *config, driverName string) {
+	fullModeSupported := func() bool {
+		unsupportedDrivers := []string{"sqlserver", "oracle"}
+		for _, dr := range unsupportedDrivers {
+			if dr == driverName {
```

Comment:
this looks a bit brittle, tell me if I'm wrong, but it looks like `driverName` is just a key used to retrieve it later and/or make sure init is only done once, and it can take pretty much any value ?

Reply:
Obviously take my perspective with a grain of salt since I'm not an owner/maintainer for the go tracer but I agree that the driverName based logic is somewhat brittle. I know it's done in other places but I wonder if we could still consider doing type checking on the `driver.Driver` or `driver.Connector`. 

Something like 
```
if connector.Driver().(go_ora.OracleDriver) {
// we know this is one of the oracle drivers
}
```

Reply:
I've added a new way to detect the driver, following this logic:
1. check driver package path using reflection
2. check dsn prefix
3. lastly check the driverName like I was doing in this version

please let me know what you think @vandonr @alexandre-normand